### ISSUE TITLE
Isar does not support non-latin characters in the directory path

### ISSUE THREAD

**Issue by lestar348** [isar/isar#878]
### Steps to Reproduce

Provide path to Isar instance with contain non-latin characters
After that get exception when Isar.open :

```
thread 'isarworker' panicked at 'called `Result::unwrap()` on an `Err` value: Utf8Error { valid_up_to: 0, error_len: Some(1) }', packages\isar_core\src\mdbx\mod.rs:50:39
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Lost connection to device.
```

### Details

 - Platform: Windows.
 - Flutter version: 3.3.3
 - Isar version: 3.0.2

**Comment by Macacoazul01** [comment]
I don't know if it helps something here but the same error was fixed on flutter 3.3.3:

https://github.com/flutter/flutter/issues/107590

**Comment by lestar348** [comment]
> I don't know if it helps something here but the same error was fixed on flutter 3.3.3:
> 
> [flutter/flutter#107590](https://github.com/flutter/flutter/issues/107590)

As far as I understand this is a rust language exception that is caused by the isar library.

**Comment by Macacoazul01** [comment]
But the fix there was changing the possible characters on the path from ascii to utf8 (and I don't think this is language exclusive)

**Comment by leisim** [comment]
The issue here is Windows. It uses UTF16 which causes issues on the Rust side

**Comment by leisim** [comment]
This is fixed in v3.0.4
